File: aula12/socket_1/cliente.py
import socket

HOST = '127.0.0.1'
PORT = 4999

if __name__ == "__main__":
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as client:
        try:
            client.connect((HOST, PORT))
            print(f"Conectado ao servidor {HOST}:{PORT}")

            while True:
                nome = input("Digite um nome (ou 'sair' para encerrar): ")
                if nome.lower() == 'sair':
                    break
                client.sendall(nome.encode('utf-8'))

        except ConnectionRefusedError:
            print("Não foi possível conectar ao servidor. Verifique se está em execução.")

File: aula12/socket_2/cliente.rb
# cliente_ruby.rb
require 'socket'

host = '127.0.0.1'
port = 5000

begin
  socket = TCPSocket.new(host, port)
  print "Digite um nome para enviar ao servidor: "
  nome = gets.chomp
  socket.puts(nome)
  puts "Nome enviado com sucesso!"
ensure
  socket.close if socket
end
